include/cell/copy/global_to_register.hpp

```diff
@@ -1,83 +1,190 @@
-            int row = row_offset + i * tile_size + (lane_id / 4);
+
+        const DType* data;
 #pragma unroll
-            for (int j = 0; j < kWidth; ++j) {
-                int col = j * tile_size + (lane_id % 4);
-                dst(i, j)(0, 0) = src(row + 0, col + 0);
-                dst(i, j)(0, 1) = src(row + 0, col + 1);
-                dst(i, j)(1, 0) = src(row + 0, col + 8);
-                dst(i, j)(1, 1) = src(row + 0, col + 9);
-            }
+        for (int i = 0; i < kRowExec; ++i) {
+            // TODO: magic number.
+            int row = i * kTileSize + lane_id / 4;
 #pragma unroll
-            for (int j = 0; j < kWidth; ++j) {
-                int col = j * tile_size + (lane_id % 4);
-                dst(i, j)(0, 2) = src(row + 8, col + 0);
-                dst(i, j)(0, 3) = src(row + 8, col + 1);
-                dst(i, j)(1, 2) = src(row + 8, col + 8);
-                dst(i, j)(1, 3) = src(row + 8, col + 9);
+            for (int j = 0; j < kColExec; ++j) {
+                // TODO: magic number.
+                int col = j * kTileSize + (lane_id % 4) * 2;
+
+                data = src + row * Global::kRowStride + col;
+                using Loader = GlobalToRegMatLoader<Global, BaseTile,
+                                                    tl::Layout::RowMajor>;
+                Loader loader;
+                loader(data, dst(i, j));
             }
         }
     }
 };
 
-/// @brief Load tile from global memory to register tile in ColMajor layout.
-/// @tparam Global Global tile type.
-/// @tparam Reg Register tile type.
-// TODO(KuangjuX): Implement LoadImpl for ColMajor layout.
-template <typename Global, typename Reg>
-struct GlobalToRegLoaderImpl<Global, Reg, tl::Layout::ColMajor> {};
+// TODO(KuangjuX): Implement this.
+template <typename Global_, typename Reg_, const int kRowExec_,
+          const int kColExec_>
+struct GlobalToRegLoaderImpl<Global_, Reg_, kRowExec_, kColExec_,
+                             tl::Layout::ColMajor> {
+    using Global = Global_;
+    using Reg = Reg_;
+    using DType = typename Global::DType;
+    using BaseTile = typename Reg::DType;
+
+    // strides to iterate over each 16x128-bits `BaseTile`.
+    static constexpr int kTileRstride =
+        BaseTileShape<DType>::row * Global::kRowStride;
+    static constexpr int kTileCstride = BaseTileShape<DType>::col;
+
+    // row stride and col stride for a thread in a warp
+    static constexpr int kStride = traits::TraitsBase<DType>::kNumPerAccess;
+    static constexpr int kLaneRstride = Global::kRowStride;
+    static constexpr int kLaneCstride = kStride;
 
-}  // namespace details
+    static constexpr int kRowExec = kRowExec_;
+    static constexpr int kColExec = kColExec_;
+
+    DEVICE void operator()(const DType* src, Reg& dst) {}
+};
 
-template <typename Global_, typename Reg_, tl::Layout type_>
-struct GlobalToRegLoader {
+/**
+ * @brief Load memory from Global memory to Register based on the warp reuse
```

Comment:
It seems that "Load memory" here should be "Load data tile"?

Reply:
fix.